src/Model/Behavior/PasswordBehavior.php

```diff
@@ -92,12 +92,22 @@ protected function sendResetPasswordEmail($user)
     /**
      * Get the user by email or username
      *
-     * @param string $reference reference could be either an email or username
+     * @param string $reference reference could be either an email or configured username field
      * @return mixed user entity if found
      */
     protected function _getUser($reference)
     {
-        return $this->_table->findByUsernameOrEmail($reference, $reference)->first();
```

Comment:
If the column does not exist, you could override this behavior to change the query, or use a custom table implementing this protected function to provide the right query. Makes sense?